crates/fixedpointmath/src/math.rs

```diff
@@ -0,0 +1,459 @@
+
+        // Using properties of logarithms we calculate x^y: -> ln(x^y) = y *
+        // ln(x) -> e^(y * ln(x)) = x^y
+        let y_int256 = y.to_i256()?;
+
+        // Compute y*ln(x) Any overflow for x will be caught in _ln() in the
+        // initial bounds check
+        let lnx = ln(self.to_i256()?)?;
+        let mut ylnx = y_int256.wrapping_mul(lnx);
+        ylnx = ylnx.wrapping_div(one.to_i256()?);
+
+        // Calculate exp(y * ln(x)) to get x^y
+        let (sign, abs) = exp(ylnx)?.into_sign_and_abs();
+        let raw = T::from_u256(abs)?.flip_sign_if(sign.is_negative());
+        Self::try_from(raw)
+    }
+}
+
+impl<T: FixedPointValue> Neg for FixedPoint<T> {
+    type Output = Self;
+
+    fn neg(self) -> Self {
+        Self::from(self.raw().flip_sign())
+    }
+}
+
+/// Takes a mapping of operator traits to `FixedPoint` methods and implements
+/// the operator and assignment operator for each one.
+macro_rules! impl_mapped_operator {
+    ($($trait:ident => $fn:ident),*) => {
+        $(
+            paste::paste! {
+
+                impl<T: FixedPointValue, U: FixedPointValue> std::ops::$trait<FixedPoint<U>> for FixedPoint<T> {
+                    type Output = Self;
+
+                    fn [<$trait:lower>](self, other: FixedPoint<U>) -> Self::Output {
+                        self.$fn(other).unwrap()
+                    }
+                }
+
+                impl<T: FixedPointValue, U: FixedPointValue> std::ops::[<$trait Assign>]<FixedPoint<U>> for FixedPoint<T> {
+                    fn [<$trait:lower _assign>](&mut self, other: FixedPoint<U>) {
+                        *self = self.[<$trait:lower>](other);
+                    }
+                }
+            }
+        )*
+    };
+    ($($tt:tt)*) => {};
+}
+
+impl_mapped_operator!(
+    // use `mul_down` for `*` and `*=`.
+    Mul => mul_down,
+    // use `div_down` for `/` and `/=`.
+    Div => div_down
+);
+
+/// Takes a list of operator traits and implements the operator and assignment
+/// operator for each one by forwarding to the corresponding method on the
+/// underlying `FixedPointValue`.
+macro_rules! impl_forwarded_operator {
+    ($($trait:ident),*) => {
+        $(
+            paste::paste! {
+
+                impl<T: FixedPointValue> std::ops::$trait for FixedPoint<T> {
+                    type Output = Self;
+
+                    fn [<$trait:lower>](self, other: Self) -> Self::Output {
+                        Self::from(self.raw().[<$trait:lower>](other.raw()))
+                    }
+                }
+
+                impl<T: FixedPointValue> std::ops::[<$trait Assign>] for FixedPoint<T> {
+                    fn [<$trait:lower _assign>](&mut self, other: Self) {
+                        *self = self.[<$trait:lower>](other);
+                    }
+                }
+            }
+        )*
+    };
+    ($($tt:tt)*) => {};
+}
+
+// Forward these operators to the underlying `FixedPointValue`.
+impl_forwarded_operator!(Add, Sub, Rem);
```

Comment:
As a consequence of forwarding these operators, they will only work with 2 FixedPoint instances that have the same underlying value type. In the future, we can implement functions to make them more flexible like the rest of the ops.